tools/src/main/java/org/apache/kafka/trogdor/workload/SustainedConnectionWorker.java

```diff
@@ -0,0 +1,563 @@
+
+                    // Create the admin client connection.
+                    this.client = AdminClient.create(this.props);
+                }
+
+                // Fetch some metadata to keep the connection alive.
+                this.client.describeCluster().nodes().get();
+
+            } catch (Throwable e) {
+                // Set the admin client to be recreated on the next cycle.
+                this.closeQuietly();
+
+                // Housekeeping to track the number of opened connections and failed connection attempts.
+                SustainedConnectionWorker.this.totalMetadataConnections.decrementAndGet();
+                SustainedConnectionWorker.this.totalMetadataFailedConnections.incrementAndGet();
+                SustainedConnectionWorker.log.error("Error while refreshing sustained AdminClient connection", e.getMessage());
+            }
+
+            // Schedule this again and set to not in use.
+            this.nextUpdate = System.currentTimeMillis() + this.refreshRate;
+            this.inUse = false;
+        }
+
+        @Override
+        protected void closeQuietly() {
+            Utils.closeQuietly(this.client, "AdminClient");
+            this.client = null;
+        }
+    }
+
+    private class ProducerSustainedConnection extends ClaimableConnection {
+
+        // These variables are used to maintain the connection itself.
+        private KafkaProducer<byte[], byte[]> producer;
+        private List<TopicPartition> partitions;
+        private String topicName;
+        private Iterator<TopicPartition> partitionsIterator;
+        private PayloadIterator keys;
+        private PayloadIterator values;
+
+        // This variable is used to maintain the scheduling of the connection keep-alive.
+        private long refreshRate;
+
+        // This variable is used to maintain the connection properties.
+        private final Properties props;
+
+        ProducerSustainedConnection() {
+
+            // These variables are used to maintain the connection itself.
+            this.producer = null;
+            this.partitions = null;
+            this.topicName = SustainedConnectionWorker.this.spec.topicName();
+            this.partitionsIterator = null;
+            this.keys = new PayloadIterator(SustainedConnectionWorker.this.spec.keyGenerator());
+            this.values = new PayloadIterator(SustainedConnectionWorker.this.spec.valueGenerator());
+
+            // This variable is used to maintain the scheduling of the connection keep-alive.
+            this.refreshRate = SustainedConnectionWorker.this.spec.refreshRateMs();
+
+            // This variable is used to maintain the connection properties.
+            this.props = new Properties();
+            this.props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, SustainedConnectionWorker.this.spec.bootstrapServers());
+            this.props.put(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION, 1);
```

Comment:
Do we need this config?
`producer.send(record).get();` ensures we get a response from the request so I don't see the value in the config

Reply:
Done.